File: tests/fixtures/openai-cua/openai-cua-retry-transient-turn-errors-fixed/catch-calls-retry-helper.ts
// Looks like the broken shape (no surrounding loop), but the catch block
// delegates to a dedicated retry helper instead of ending the run directly.
export async function runTurn(client: any, model: string, input: any[]) {
  try {
    return await client.responses.create({ model, input });
  } catch (exc) {
    return retryTurnWithBackoff(client, model, input, exc);
  }
}

async function retryTurnWithBackoff(client: any, model: string, input: any[], lastError: unknown) {
  await new Promise((resolve) => setTimeout(resolve, 300));
  return client.responses.create({ model, input });
}

File: tests/fixtures/openai-cua/openai-cua-retry-transient-turn-errors-fixed/catch_calls_retry_helper.py
def run_turn(client, input_items):
    try:
        return client.responses.create(model="computer-use-preview", input=input_items)
    except Exception as exc:
        return retry_turn(client, input_items, exc)


def retry_turn(client, input_items, original_error):
    return client.responses.create(model="computer-use-preview", input=input_items)
